tests/src/string/src/main.sw

```diff
@@ -130,116 +133,107 @@ impl StringTest for Contract {
-        // let mut bytes = Bytes::new();
-
-        // bytes.push(NUMBER0);
-        // bytes.push(NUMBER1);
-        // bytes.push(NUMBER2);
-        // bytes.push(NUMBER3);
-        // bytes.push(NUMBER4);
-
-        // let raw_slice = bytes.as_raw_slice();
-        // let string_from_slice = String::from(raw_slice);
-        // assert(bytes.len() == string_from_slice.len());
-        // assert(bytes.capacity() == string_from_slice.capacity());
-        // assert(bytes.get(0).unwrap() == string_from_slice.nth(0).unwrap());
-        // assert(bytes.get(1).unwrap() == string_from_slice.nth(1).unwrap());
-        // assert(bytes.get(2).unwrap() == string_from_slice.nth(2).unwrap());
-}
+        let mut bytes = Bytes::new();
+
+        bytes.push(NUMBER0);
+        bytes.push(NUMBER1);
+        bytes.push(NUMBER2);
+        bytes.push(NUMBER3);
+        bytes.push(NUMBER4);
+
+        let raw_slice = bytes.as_raw_slice();
+        let mut string_from_slice = String::from_raw_slice(raw_slice);
+        assert(bytes.len() == string_from_slice.len());
+        assert(bytes.get(0).unwrap() == string_from_slice.nth(0).unwrap());
+        assert(bytes.get(1).unwrap() == string_from_slice.nth(1).unwrap());
+        assert(bytes.get(2).unwrap() == string_from_slice.nth(2).unwrap());
+    }
 
     fn test_from_utf8() {
-
-        // Uncomment when https://github.com/FuelLabs/sway/issues/4408 is resolved
-        // let mut vec: Vec<u8> = Vec::new();
-
-        // vec.push(NUMBER0);
-        // vec.push(NUMBER1);
-        // vec.push(NUMBER2);
-        // vec.push(NUMBER3);
-        // vec.push(NUMBER4);
-
-        // let string_from_uft8 = String::from_utf8(vec);
-        // assert(vec.len() == string_from_uft8.len());
-        // assert(vec.capacity() == string_from_uft8.capacity());
-        // assert(vec.get(0).unwrap() == string_from_uft8.nth(0).unwrap());
-        // assert(vec.get(1).unwrap() == string_from_uft8.nth(1).unwrap());
-        // assert(vec.get(2).unwrap() == string_from_uft8.nth(2).unwrap());
-}
+        let mut vec: Vec<u8> = Vec::new();
+
+        vec.push(NUMBER0);
+        vec.push(NUMBER1);
+        vec.push(NUMBER2);
+        vec.push(NUMBER3);
+        vec.push(NUMBER4);
+
+        let mut string_from_uft8 = String::from_utf8(vec);
+        assert(vec.len() == string_from_uft8.len());
+        assert(vec.capacity() == string_from_uft8.capacity());
+        assert(vec.get(0).unwrap() == string_from_uft8.nth(0).unwrap());
+        assert(vec.get(1).unwrap() == string_from_uft8.nth(1).unwrap());
+        assert(vec.get(2).unwrap() == string_from_uft8.nth(2).unwrap());
+    }
 
     fn test_insert() {
         let mut string = String::new();
 
         assert(string.len() == 0);
 
-        // Uncomment when https://github.com/FuelLabs/sway/issues/4408 is resolved
-        // string.insert(NUMBER0, 0);
-        // assert(string.len() == 1);
-        // assert(string.nth(0).unwrap() == NUMBER0);
+        string.insert(NUMBER0, 0);
```

Comment:
I would fill the string with a few values and then insert over them while asserting that they have changed after each insertion rather than mixing up inserting, pushing, asserting with various offsets